contrib/mesos/pkg/scheduler/podtask/predicate.go

```diff
@@ -57,21 +61,29 @@ func NodeSelectorPredicate(t *T, offer *mesos.Offer) bool {
 
 	// check the NodeSelector
 	if len(t.Pod.Spec.NodeSelector) > 0 {
-		slaveLabels := map[string]string{}
-		for _, a := range offer.Attributes {
-			if a.GetType() == mesos.Value_TEXT {
-				slaveLabels[a.GetName()] = a.GetText().GetValue()
+		l := map[string]string{
+			"kubernetes.io/hostname": offer.GetHostname(),
+		}
+		if n != nil && n.Labels != nil {
+			for k, v := range n.Labels {
+				if !strings.HasPrefix(k, node.LabelPrefix) {
```

Comment:
I still don't understand why we're checking prefix here. If we're *really* concerned about leakage (and I don't know why we would be since the `SlaveAttributesToLabels` call overrides any duplicate keys) I'd rather replace this `string.HasPrefix` call with a call into `node` along the lines of `if !node.IsSlaveAttrLabel(...)` and make `LabelPrefix` package private.